content/200-orm/200-prisma-client/500-deployment/301-edge/450-deploy-to-cloudflare.mdx

```diff
@@ -0,0 +1,526 @@
+
+> **Note**: There's [also work being done](https://github.com/sidorares/node-mysql2/pull/2289) on the `node-mysql2` driver which will enable to access traditional MySQL databases from edge functions as well.
+
+### Setting your database connection URL as an environment variable
+
+First, ensure that the `DATABASE_URL` is set as the `url` of the `datasource` in your Prisma schema:
+
+```prisma
+datasource db {
+  provider = "postgresql" // this might also be `mysql` or another value depending on your database
+  url      = env("DATABASE_URL")
+}
+```
+
+#### Development
+
+When using your Worker in **development**, you can configure your database connection via the [`.dev.vars` file](https://developers.cloudflare.com/workers/configuration/secrets/#secrets-in-development) locally.
+
+Assuming you use the `DATABASE_URL` environment variable from above, you can set it inside `.dev.vars` as follows:
+
+```bash
+# .dev.vars
+DATABASE_URL="your-database-connection-string"
+```
+
+In the above snippet, `your-database-connection-string` is a placeholder that you need to replace with the value of your own connection string, for example:
+
+```bash
+# .dev.vars
+DATABASE_URL="postgresql://admin:mypassword42@somehost.aws.com:5432/mydb"
+```
+
+Note that the `.dev.vars` file is not compatible with `.env` files which are typically used by Prisma ORM.
+
+This means that you need make sure that Prisma ORM gets access to the environemnt variable when needed, e.g. when running a Prisma CLI command like `prisma migrate dev`.
+
+There are several options for achieving this:
+
+- Run your Prisma CLI commands using `dotenv` to specify from where the CLI should read the environment variable, for example:
+  ```terminal
+  dotenv -e .dev.vars -- npx prisma migrate dev
+  ```
+- Create a script in `package.json` that reads `.dev.vars` via `dotenv`. You can then execute `prisma` commands as follows: `npm run env -- npx prisma migrate dev`. Here's a reference for the script:
+  ```js
+  "scripts":  { "env": "dotenv -e .dev.vars" }
+  ```
+- Duplicate the `DATABASE_URL` and any other relevant env vars into a new file called `.env`. that's used by Prisma ORM.
+
+#### Production
+
+When deploying your Worker to **production**, you'll need to set the database connection using the `wrangler` CLI:
+
+```terminal
+npx wrangler secret put DATABASE_URL
+```
+
+The command is interactive and will ask you to enter the value for the `DATABASE_URL` env var as the next step in the terminal.
+
+> **Note**: This command requires you to be authenticated, and will ask you to log in to your Cloudflare account in case you are not.
+
+### Size limits on free accounts
+
+Cloudflare has a [size limit of 1 MB on free accounts](https://developers.cloudflare.com/workers/platform/limits/). If your application bundle with Prisma ORM exceeds that size, we recommend upgrading to a paid account or using Prisma Accelerate to deploy yout application.
```

Comment:
```suggestion
Cloudflare has a [size limit of 1 MB for Workers on the free plan](https://developers.cloudflare.com/workers/platform/limits/). If your application bundle with Prisma ORM exceeds that size, we recommend upgrading to a paid Worker plan or using Prisma Accelerate to deploy your application.
```
Because Cloudflare account is different from a Worker plan.
Example https://developers.cloudflare.com/workers/platform/pricing/